File: face/geomajas-face-dojo/client/src/main/resources/geomajas/spatial/geometry/util/GeometryOperation.js
/*
 * This is part of Geomajas, a GIS framework, http://www.geomajas.org/.
 *
 * Copyright 2008-2011 Geosparc nv, http://www.geosparc.com/, Belgium.
 *
 * The program is available in open source according to the GNU Affero
 * General Public License. All contributions in this program are covered
 * by the Geomajas Contributors License Agreement. For full licensing
 * details, see LICENSE.txt in the project root.
 */

dojo.provide("geomajas.spatial.geometry.util.GeometryOperation");
dojo.declare("GeometryOperation", null, {

	/**
	 * @fileoverview Interface for a geometric operation.
	 * @class General interface for geometric opertations.
	 * @author Pieter De Graef
	 * @constructor
	 */
	constructor : function () {
	},

	/**
	 * The main edit function. Is passed a geometry object. If other values are
	 * needed, pass them through the constructor, or via setters. Does not
	 * return anything, so it adjusts the given geometry!
	 * @param geometry The {@link Geometry} object to be adjusted.
	 */
	edit : function (geometry) {
		alert("GeometryOperartion.edit : implement me!");
	}
});

File: client/geomajas-client-smartgwt/client/src/main/java/org/geomajas/smartgwt/client/spatial/geometry/operation/GeometryOperation.java
/*
 * This is part of Geomajas, a GIS framework, http://www.geomajas.org/.
 *
 * Copyright 2008-2013 Geosparc nv, http://www.geosparc.com/, Belgium.
 *
 * The program is available in open source according to the GNU Affero
 * General Public License. All contributions in this program are covered
 * by the Geomajas Contributors License Agreement. For full licensing
 * details, see LICENSE.txt in the project root.
 */

package org.geomajas.smartgwt.client.spatial.geometry.operation;

import org.geomajas.annotation.Api;
import org.geomajas.smartgwt.client.spatial.geometry.Geometry;

/**
 * <p>
 * General interface for operations on Geometry objects. All operations should return a new Geometry instead of changing
 * the original.
 * </p>
 *
 * @author Pieter De Graef
 * @since 1.6.0
 */
// @extract-start GeometryOperation, GeometryOperation interface
@Api(allMethods = true)
public interface GeometryOperation {

	/**
	 * The main edit function. It is passed a geometry object. If other values are needed, pass them through the
	 * constructor, or via setters.
	 *
	 * @param geometry
	 *            The {@link org.geomajas.smartgwt.client.spatial.geometry.Geometry} object to be adjusted.
	 * @return Returns the resulting geometry, leaving the original unharmed.
	 */
	Geometry execute(Geometry geometry);
}
// @extract-end
